### ISSUE TITLE
Aliases with embed:rst:leading-asterisk no longer work in Doxygen 1.12

### ISSUE THREAD

**Issue by TruongMinh89** [doxygen/doxygen#11172]
**Describe the bug**
[example3.zip](https://github.com/user-attachments/files/17260665/example3.zip)

In Doxygen 1.9.7, I could use the following aliases to embed reStructuredText (RST) content with leading asterisks removed:
```
ALIASES += "rst=\verbatim embed:rst:leading-asterisk"
ALIASES += "endrst=\endverbatim"
```

However, in Doxygen 1.12, this no longer works as expected. The issue occurs mainly when aliases are used within this RST block. The content is not rendered correctly, and any alias inside the rst block is ignored. 

**Screenshots**
![image](https://github.com/user-attachments/assets/35302583-30fd-4684-822e-5fca4b45878a)

**To Reproduce**
Use
```
ALIASES += "rst=\verbatim embed:rst:leading-asterisk"
ALIASES += "endrst=\endverbatim"
```
and use any aliases within.

**Expected behavior**
Aliases within rst block should be rendered correctly

**Version**
Doxygen version used: 1.12.0

**Stack trace**
N/A

**Additional context**
N/A

**Comment by albert-github** [comment]
In don't see any example with which we could try to reproduce the problem, so:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by TruongMinh89** [comment]
I have added the example.zip:
- Doxyfile: Contain relevant ALIASES only. I hope that you can add it to general Doxyfile to test that.
- An example .c file that used those ALIASES

**Comment by TruongMinh89** [comment]
Sorry, have to update the example several times.

**Comment by albert-github** [comment]
I looked at the example from https://github.com/doxygen/doxygen/issues/11172#issuecomment-2393930233 
When running this example with doxygen 1.9.7 I get:
```
error: Illegal ALIASES format 'csv_table{3::}=.. csv-table:: \1^^\3   :header: \2^^'. Use "name=value" or "name{n}=value", where n is the number of arguments
```
which is logical as the "Allow custom separator for ALIASES" were introduced in doxygen 1.9.8

- So this is not the right definition for the 1.9.7 version.

When running the current doxygen 1.12.0 version and using the debug option `commentcnv` (so `doxygen -d commentcnv`) we see:
```
...
/**
 * All outputs request shall be interpreted as physically zero after
 * initialization.
 *
 * \verbatim embed:rst:leading-asterisk
 * @csv_table{Frozen Delights!::Treat, Quantity, Description::}
 *    Albatross, 2.99, On a stick!
 *    Crunchy Frog, 1.49, If we took thsadasdasdas asdadssadd
 *    Gannet Ripple, 2, 3
 * \endverbatim
 *
 */
...
```

In other words the the `@rst` is evaluated and at that moment we are in a verbatim block and this means that the `@csv_table` is not handled anymore (apparently the `@endrst` is handled, @doxygen ?). 

This is all due to the improved handling of the `verbatim` command.

I see the following possible solutions:
- Remove the `@rst` from the `.c` file and change the `csv_table` ALIASES into
  ```
  ALIASES += "csv_table{3::}=@rst^^.. csv-table:: \1^^\3   :header: \2^^"
  ```
- another way would be to use the internal doxygen commands `iverbatim` and `endiverbatim` in the `ALIASES` instead of the `verbatim` and `endverbatim` commands (see the chapter "Overview of introduction of internal special commands" in the internal documentation https://doxygen.github.io/doxygen-docs/d6/d57/pg_cmds_int.html)

Both look like they give the correct solution.

@doxygen do you have another idea? as I'm not really fond of either solution for the end user

**Comment by TruongMinh89** [comment]
Thanks for the support!

- For the first solution: I found it difficult to apply when there’s more than a single alias in the RST block. For example, it becomes unclear how to handle something like this:
`* @rst
* @csv_table{Frozen Delights!::Treat, Quantity, Description::}
*    Albatross, 2.99, On a stick!
*    Crunchy Frog, 1.49, If we took thsadasdasdas asdadssadd
*    Gannet Ripple, @doxylink{MicroSlip_TestInit1}, @doxylink{MicroSlip_CnctStateType}
* @endrst
`
Combining multiple aliases within the same RST block makes the first solution harder to implement.

- For the second solution: I encountered an issue with the indentation—specifically, before :header:, only a single space is created instead of the expected 3 spaces, which affects formatting

**Comment by albert-github** [comment]
Regarding the second solution:
What happens whe you replace the alias:
```
ALIASES += "csv_table{3::}=.. csv-table:: \1^^\3   :header: \2^^"
```
by
```
ALIASES += "csv_table{3::}=.. csv-table:: \1^^\3\   :header: \2^^"
```
or
```
ALIASES += "csv_table{3::}=.. csv-table:: \1^^\3&nbsp;  :header: \2^^"
```

**Comment by albert-github** [comment]
Regarding the first solution:
It will indeed be a bit problematic when a "replaced" `ALIASES` is used multiple times in a one block,
In the given example though I think only `@csv_table` has to be changed and `@doxylink` can remain as it is (though I have no insight in what `@doxylink` exactly does0.

**Comment by TruongMinh89** [comment]
![image](https://github.com/user-attachments/assets/cee3e57f-465f-417d-b197-2ee03dfcd493)
[example4.zip](https://github.com/user-attachments/files/17274814/example4.zip)

I've tried those but haven't had good results yet.

**Comment by albert-github** [comment]
What happens when you use:
```
ALIASES += "csv_table{3::}=.. csv-table:: \1\n\3   :header: \2^^"
```

**Comment by albert-github** [comment]
what happens when you use:
```
ALIASES += "csv_table{3::}=.. csv-table:: \1^^\3     :header: \2^^\3     :class: sphinx-datatable^^"
```
(Note the number of spaces after `\3`)


How should the output look like?

**Comment by TruongMinh89** [comment]
For this one;
```
ALIASES += "csv_table{3::}=.. csv-table:: \1\n\3   :header: \2^^"
```

```
 embed:rst:leading-asterisk
  .. csv-table:: Frozen Delights!
   :header: Treat, Quantity, Description

     Albatross, 2.99, On a stick!
     Crunchy Frog, 1.49, If we took thsadasdasdas asdadssadd
     Gannet Ripple, :doxygen:`MicroSlip_TestInit1`, :doxygen:`MicroSlip_CnctStateType`
```
Although the :header: only has ident 1 space, while data rows have ident 3 spaces
But this time it still get the ok result
![image](https://github.com/user-attachments/assets/6088932d-5774-4408-bf81-4223d59c10a5)

But it is not ok with more complex example (all aliases for the directives use 3 spaces ident)
```
/**
 *
 * @rst
 * @swut{MicroSlip_TestAdapterOutput2::SWS_MicroSlip_003, SWD_MicroSlip_AdapterOutput}
 *    @csv_table3{Test Steps::Step, Step Description, Expected Result::   }
 *       1, "| Check voltage 1
 *       | Check @doxylink{MicroSlip_TestInit1}", "Voltages is 12.8V"
 *       2, "- dssadsda
 *       - dsadsadsadsdsa
 *       - asdassda", asddsdsa 1 item 6
 *       3, "Check
 *       voltage 1", "- sdsdasd
 *       - adsdsadsa"
 *
 *    @teststeps_csvtable{   }
 *       1, "| Check voltage 1
 *       | Check @doxylink{MicroSlip_TestInit1}", "Voltages is 12.8V"
 *       2, "- dssadsda
 *       - dsadsadsadsdsa
 *       - asdassda", asddsdsa 1 item 6
 *       3, "Check
 *       voltage 1", "- sdsdasd
 *       - adsdsadsa"
 *
 *    @teststeps_listtable{   }
 *       * - 1. check m_tControlMode and m_tCycleControlModeRequest
 *         - | - Set m_tControlMode and m_tCycleControlModeRequest = test loop value equal to each value in CtrlMode_t
 *           | - Call under-tested function
 *         - | - m_tControlMode and m_tCycleControlModeRequest should be equal to test loop value.
 * @endrst
 */
```
Result:
```
 embed:rst:leading-asterisk
.. swut::
 :id: SWUT_MicroSlip_TestAdapterOutput2
 :reports: [[allocate_testreports()]]
 :testresult: [[get_testresult()]]
 :tests: SWS_MicroSlip_003, SWD_MicroSlip_AdapterOutput
 
   .. csv-table:: Test Steps
    :header: Step, Step Description, Expected Result
    :class: sphinx-datatable
 
      1, "| Check voltage 1
      | Check :doxygen:`MicroSlip_TestInit1`", "Voltages is 12.8V"
      2, "- dssadsda
      - dsadsadsadsdsa
      - asdassda", asddsdsa 1 item 6
      3, "Check
      voltage 1", "- sdsdasd
      - adsdsadsa"

   .. csv-table:: Test Steps
    :header: **Step**, **Step Description**, **Expected Result**
    :class: sphinx-datatable
 
      1, "| Check voltage 1
      | Check :doxygen:`MicroSlip_TestInit1`", "Voltages is 12.8V"
      2, "- dssadsda
      - dsadsadsadsdsa
      - asdassda", asddsdsa 1 item 6
      3, "Check
      voltage 1", "- sdsdasd
      - adsdsadsa"

   .. list-table:: Test Steps
    :header-rows: 1
    :widths: auto
    :class: sphinx-datatable
  
       * - **Step**
      - **Step Description**
      - **Expected Result** 
      * - 1. check m_tControlMode and m_tCycleControlModeRequest
        - | - Set m_tControlMode and m_tCycleControlModeRequest = test loop value equal to each value in CtrlMode_t
          | - Call under-tested function
        - | - m_tControlMode and m_tCycleControlModeRequest should be equal to test loop value.
```
![image](https://github.com/user-attachments/assets/43806b57-f615-4b5e-9ae1-ca7469e7b23d)


With 5 spaces ident used in the aliases as you mentioned, the results are the same with 3 spaces ident.

**Comment by TruongMinh89** [comment]
Above results is from a .C file
But if I do the same block comment using % in a Cpp file then (same if 3 spaces or 5 spaces), don't know why:
```
 embed:rst:leading-asterisk
.. csv-table:: Frozen Delights!
    :header: Treat, Quantity, Description
    :class: sphinx-datatable
 
   Albatross, 2.99, On a stick!
   Crunchy Frog, 1.49, If we took thsadasdasdas asdadssadd
   Gannet Ripple, :doxygen:`MicroSlip_TestInit1`, :doxygen:`MicroSlip_CnctStateType`
```

**Comment by albert-github** [comment]
Seen the result in https://github.com/doxygen/doxygen/issues/11172#issuecomment-2396636166 looks like your problem is solved (correct?). Does the complex example work as well?
- which doxygen results are you using to generate the results?
- how do you display the rst results?

Maybe we should have a look why the `.c` file gives a different result from the `.cpp` file, so to have a good look:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

The change regarding `verbatim` is by design so I doubt this will be changed back.

**Comment by TruongMinh89** [comment]
For complex example you can see in rendered html, we have lost the last table.
In C, it generate 1 spaces, but in Cpp it is 4 spaces. And in both files, the complex example didn't work (Need ident 3 spaces output)

I have used doxysphinx to create RST files from the Doxygen-generated HTML files and tagfile.xml. Then, I built those RST files using Sphinx.

Let me prepare the example.

**Comment by TruongMinh89** [comment]
[example4.zip](https://github.com/user-attachments/files/17278034/example4.zip)
Please find attached file for the example.

**Comment by albert-github** [comment]
I get warnings:
```
classDocumentationExample.cpp:36: warning: Invalid list item found
classDocumentationExample.cpp:41: warning: unexpected command endiverbatim
```

- Is there a `@rst` missing?

Also the code in the `.c` file is not valid code (see `MicroSlip_TestAdapterOutput1()`).


The difference between the `.c` and the `.cpp`, amazingly, looks like a result of of the initial space before the  `/**` / `*` lines.

I'll try to dive a bit in this space problem.

**Comment by TruongMinh89** [comment]
You are right. I'm sorry I didn't prepare the example well. Added @rst for cpp and correct MicroSlip_TestAdapterOutput1 (removed one `{`)
[example5.zip](https://github.com/user-attachments/files/17279124/example5.zip)

**Comment by TruongMinh89** [comment]
![image](https://github.com/user-attachments/assets/b9da3bc5-dba0-4472-9b57-3235b748e61e)
Result from generated Doxygen html files

**Comment by albert-github** [comment]
As noted the problem is due to the number of spaces with which the commentblock is indented (i.e. the spaces before the `/**` / *`).
It looks like a subtle interaction between the replacement of the `^^` into `\ilinebr` (in aliases.cpp during the comment cnv phase) and the `stripIndentation` (in util.cpp during the comment scan phase).

The `ALIASES` are read and the `^^` are replaced in an early phase of doxygen, during the reading of the configuration file, and here the indentation is not yet known, so this information is not taken into consideration.

@doxygen do you see a solution?

**Comment by TruongMinh89** [comment]
Thanks for the clarification.

**Comment by TruongMinh89** [comment]
Can we label this? @albert-github

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #11175

**Comment by albert-github** [comment]
@TruongMinh89 please test the proposed patch (as far as you aren't already doing it) and report back.

**Comment by TruongMinh89** [comment]
I have put one finding for the PR. Tested and should be ok if we don't have additional space created

**Comment by TruongMinh89** [comment]
Hi, I tested and it work ok with C and Cpp. What needed is to change from 3 spaces to 2 spaces ident in the aliases
But it does not work with python file

```
##
# All outputs request shall be interpreted as physically zero after
# initialization.
#
# @rst
# @csv_table{Frozen Delights!::Treat, Quantity, Description::}
#    Albatross, 2.99, On a stick!
#    Crunchy Frog, 1.49, If we took thsadasdasdas asdadssadd
#    Gannet Ripple, @doxylink{MicroSlip_TestInit1}, @doxylink{MicroSlip_CnctStateType}
# @endrst
#
def test_function():
    return
```

Result:
```
 embed:rst:leading-asterisk
 .. csv-table:: Frozen Delights!@ilinebr   :header: Treat, Quantity, Description@ilinebr   :class: sphinx-datatable@ilinebr 
    Albatross, 2.99, On a stick!
    Crunchy Frog, 1.49, If we took thsadasdasdas asdadssadd
    Gannet Ripple, :doxygen:`MicroSlip_TestInit1`, :doxygen:`MicroSlip_CnctStateType`
```

**Comment by albert-github** [comment]
I don't think the example is correct as it has no spaces in front of the `#`. When adding these I can reproduce the problem. Also there is no settings file so always:
- attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

Adding the strip function to python (and Fortran) should solve the issue. 
I've pushed a commit to #11175

**Comment by TruongMinh89** [comment]
Thanks for you remark! I will note that.

**Comment by TruongMinh89** [comment]
Tested with latest source code and with this example:
[example_6.tar.gz](https://github.com/user-attachments/files/17364062/example_6.tar.gz)

Why we have different result between csv_table1 and csv_table2?

**Comment by albert-github** [comment]
In the output we see:
![image](https://github.com/user-attachments/assets/d2665914-5345-4cae-9e16-3c2eb47d1a41)

Not the position of the first `:` of `:header:`.

In your code we see:
```
     *    @csv_table1{::head1, head2, head3::   }
     *       val1, val2, val3
     *
     *    @csv_table2{::head1, head2, head3::   }
     *       val1, val2, val3
```
and in the `ALIASES` we see
```
ALIASES += "csv_table1{3::}=.. csv-table:: \1^^\3   :header: \2^^"

ALIASES += "csv_table2{3::}=.. csv-table:: \1^^   \3:header: \2^^"
```
at the moment of reading the `ALIASES` and expanding the `^^` nothing is known about the `\3` and thus `^^\` is see as `^^` followed by some character (which might lead to problems indicated in "otherwise the command `ilinebr` cannot be correctly determined e.g. when a "normal" word is following") and thus replaced by `\ilinebr \`. for `^^ ` we see the space and thus it is replaced by `\ilinebr ` as a consequence of the remark by doxygen in https://github.com/doxygen/doxygen/pull/11175#discussion_r1797642886).


(small side not in the example there is a mixture of `<tabs>` and space which might give strange results without setting the `TAB_SIZE`. Better is not to use `<tabs>` at all)

**Comment by TruongMinh89** [comment]
I though `\3` replaced by `   ` before processing the escape `^^`.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
